std/functional.d

```diff
@@ -899,6 +903,149 @@ template partial(alias fun, alias arg)
     assert(result2 == 1.5f);
 }
 
+/**
+Takes a function of (potentially) many arguments, and returns a function taking
+one argument and returns a callable taking the rest.  f(x, y) == curry(f)(x)(y)
+
+Params:
+    F = a function taking at least one argument
+    t = a callable object whose opCall takes at least 1 object
+Returns:
+    A single parameter callable object
+*/
+template curry(alias F)
+{
+    //inspired from the implementation from Artur Skawina here:
+    //https://forum.dlang.org/post/mailman.1626.1340110492.24740.digitalmars-d@puremagic.com
+    //This implementation stores a copy of all filled in arguments with each curried result
+    //this way, the curried functions are independent and don't share any references
+    //eg: auto fc = curry!f;  auto fc1 = fc(1); auto fc2 = fc(2); fc1(3) != fc2(3)
+    auto curry()
+    {
+        import std.traits : Parameters;
+        alias FParams = Parameters!F;
+
+        struct CurryImpl(size_t N)
+        {
+            FParams[0..N] storedArguments;
+            static if(N > 0)
+            {
+                this(const ref CurryImpl!(N-1) prev, FParams[N] arg)
+                {
+                    static foreach(i; 0..N-1)
+                    {
+                        storedArguments[i] = prev.storedArguments[i];
+                    }
+                    storedArguments[N-1] = arg;
+                }
+            }
+
+            auto opCall(FParams[N] arg) const
+            {
+                static if(N == FParams.length -1)
```

Comment:
space...

Reply:
Is there an easy way to use tooling to fix these automatically in the future?  Running the whole file through dfmt seemed like it would add lots of unnecessary changes

Reply:
Unfortunately not for Phobos as it's not formatted consistently (it would create too much churn and unrelated changes to run dfmt on entire Phobos).
There's an open issue for dfmt to run it only on the current diff, but so far this is not supported.